src/UnderlineNav2/UnderlineNavItem.tsx

```diff
@@ -70,30 +77,73 @@ export const UnderlineNavLink = forwardRef(
     const textStyles: BetterSystemStyleObject = {
       whiteSpace: 'nowrap'
     }
+
+    const wrapperStyles = {
+      display: 'inline-flex',
+      paddingY: 1,
+      paddingX: 2,
+      borderRadius: 2
+    }
     const smallVariantLinkStyles = {
-      paddingY: 2,
+      paddingY: 1,
       fontSize: 0
     }
     const defaultVariantLinkStyles = {
-      paddingY: 3,
+      paddingY: 2,
       fontSize: 1
     }
 
     const linkStyles = {
+      position: 'relative',
       display: 'inline-flex',
       color: 'fg.default',
       textAlign: 'center',
-      borderBottom: '2px solid transparent',
-      borderColor: selectedLink === ref ? 'primer.border.active' : 'transparent',
       textDecoration: 'none',
-      paddingX: 2,
-      marginRight: 3,
+      paddingX: 1,
+      borderColor: selectedLink === ref ? 'primer.border.active' : 'transparent',
       ...(variant === 'small' ? smallVariantLinkStyles : defaultVariantLinkStyles),
-      '&:hover, &:focus': {
-        borderColor: selectedLink === ref ? 'primer.border.active' : 'neutral.muted',
-        transition: '0.2s ease'
+      '&:hover > div[data-component="wrapper"] ': {
+        backgroundColor: 'neutral.muted',
+        transition: 'background .12s ease-out'
+      },
+      '&:focus': {
+        outline: 0,
+        '& > div[data-component="wrapper"]': {
+          boxShadow: `inset 0 0 0 2px #0969da`
+        },
+        // where focus-visible is supported, remove the focus box-shadow
+        '&:not(:focus-visible) > div[data-component="wrapper"]': {
+          boxShadow: 'none'
+        }
+      },
+      '&:focus-visible > div[data-component="wrapper"]': {
+        boxShadow: `inset 0 0 0 2px #0969da`
+      },
+      // renders a visibly hidden "copy" of the label in bold, reserving box space for when label becomes bold on selected
+      '& span[data-content]::before': {
+        content: 'attr(data-content)',
+        display: 'block',
+        height: 0,
+        fontWeight: '600',
+        visibility: 'hidden'
+      },
+      // selected state styles
+      '&::after': {
```

Comment:
Previous element for the border bottom was creating a jumpy affect when selected as it was conditionally rendered. So to avoid that jumpiness, used pseudo class (ref: [Primer CSS](https://github.com/primer/css/blob/main/src/navigation/underline-nav.scss#L90))